### PR TITLE
Gh 135 semgrep UI

### PR DESCRIPTION
**Pull request by rocky341**
Hi @t92549, 
PR for ticket #135 
Adds semgrep and fixes for some vulnerabilities which were identified by semgrep

### CODE REVIEW

**Comment by t92549** [comment]
@K-itKat20 Let me know when this is ready for review. Also, not sure why but this PR is showing up as 1,179 commits?

**Comment by K-itKat20** [comment]
> @K-itKat20 Let me know when this is ready for review. Also, not sure why but this PR is showing up as 1,179 commits?

@t92549 This PR is ready for review. Not sure why there's 1000+ commits showing up as only 20 files have been changed. This is somehow picking up commits from May.

**Review by t92549 — CHANGES_REQUESTED** [review]
I am getting a few eslint warnings

File: gaas-ui/.gitignore [hunk 1]
```diff
@@ -4,6 +4,7 @@
 /node_modules
 /.pnp
 .pnp.js
+package-lock.json
```
[the remark this answers is not in the document]
Comment (rocky341) [reply gaas-ui/.gitignore]: Removed from here and left in root gitignore

File: .github/workflows/semgrep.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            p/ci
+            p/jwt
+            p/nodejs
+            p/nodejsscan
+          auditOn: push
+          generateSarif: "1"
+      - name: Github Security Dashboard
+        uses: github/codeql-action/upload-sarif@v1
+        with:
+          sarif_file: semgrep.sarif
```
[the remark this answers is not in the document]
Comment (rocky341) [reply .github/workflows/semgrep.yml]: Added

File: .github/workflows/semgrep.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+name: semgrep-action
+
+on: [push, pull_request ]
```
[the remark this answers is not in the document]
Comment (rocky341) [reply .github/workflows/semgrep.yml:2]: Removed white space

File: .semgrepignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+/gaas-ui/server/middleware.js
+/gaas-ui/test
```
[the remark this answers is not in the document]
Comment (rocky341) [reply .semgrepignore]: Added newline

File: Dockerfile.gaas-ui [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+# remove the default Nginx configuration file
+RUN rm /etc/nginx/conf.d/default.conf
+# copy Nginx config file to the ui image
+COPY gaas-ui/nginx/nginx.conf /etc/nginx/conf.d
+# expose port 8081 for the Nginx server to listen to
+EXPOSE 8081
+# comment user directive as master process is run as user in OpenShift
+RUN sed -i.bak 's/^user/#user/' /etc/nginx/nginx.conf
+# run Nginx in foreground and not as daemon
+CMD ["nginx", "-g", "daemon off;"]
```
[the remark this answers is not in the document]
Comment (rocky341) [reply Dockerfile.gaas-ui]: Added newline

File: Dockerfile.gaas-ui [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# STAGE 1 - BUILD UI APP
```
[the remark this answers is not in the document]
Comment (rocky341) [reply Dockerfile.gaas-ui:1]: I think it needs to be in the root for the pipeline to work

File: .github/workflows/semgrep.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
+name: semgrep-action
+on: [push, pull_request ]
```
Comment (t92549) [inline .github/workflows/semgrep.yml:2]: Sorry I mean the white space inside `on: [push, pull_request ]`

File: gaas-ui/src/components/view-graphs/view-graphs-table.tsx [hunk 8]
```diff
@@ -146,8 +154,8 @@ function MainGraphTableRow(props: IGraphRow) {
                     }}> {graph.getConfigName().charAt(0).toUpperCase()}</Avatar></TableCell>
                 <TableCell aria-label={"graph-status"}>
                     <StatusChip status={graph.getStatus()}/></TableCell>
-                <TableCell aria-label={"graph-url"}><a href={graph.getUrl()} target="_blank"
-                                                       rel="noreferrer">{graph.getUrl()}</a></TableCell>
+                <TableCell aria-label={"graph-url"}>
+                    <a id={graph.getId()} target="_blank" rel="noopener noreferrer">{graph.getUrl()}</a></TableCell>
```
Comment (t92549) [inline gaas-ui/src/components/view-graphs/view-graphs-table.tsx:158]: Still getting es-lint warnings

**Review by t92549 — CHANGES_REQUESTED** [review]

File: .github/workflows/semgrep.yml (hunk 7 above)
Comment (rocky341) [reply .github/workflows/semgrep.yml]: Sorry should have noticed that. Updated

File: gaas-ui/src/components/view-graphs/view-graphs-table.tsx (hunk 8 above)
Comment (rocky341) [reply gaas-ui/src/components/view-graphs/view-graphs-table.tsx]: Added a placeholder href to fix the issue.

**Review by t92549 — APPROVED** [review]

**Comment by t92549** [comment]
@rocky341 Thanks for all the fixes!

**Pull request merged**